disk/check.py

```diff
@@ -289,3 +300,16 @@ def _list_devices(self, df_output):
 
         # Filter fake or unwanteddisks.
         return [d for d in flattened_devices if self._keep_device(d)]
+
+    def _compile_tag_re(self):
+        """
+        Compile regex strings from device_tag_re option and return list of compiled regex/tag pairs
+        """
+        device_tag_re = []
+        for pair in self._device_tag_re:
+            for regex_str, tag in pair.items():
+                try:
+                    device_tag_re.append([re.compile(regex_str), [t.strip() for t in tag.split(",")]])
+                except TypeError:
+                    self.log.warning('{0} is not a valid regular expression and will be ignored'.format(regex_str))
+        self._device_tag_re = device_tag_re
```

Comment:
You should avoid using the same variable for the configuration data and post-processing data. It make the code less obvious.

Reply:
I meant that `_device_tag_re` before the function contains the raw config data and after the precompiled regexp. We should store the final data in another variable the the one used to store the raw contfig.